Mathlib/Algebra/Star/Unitary.lean

```diff
@@ -133,12 +138,26 @@ theorem _root_.IsUnit.mem_unitary_of_mul_star_self {u : R} (hu : IsUnit u)
   star_star u ▸
     (hu.star.mem_unitary_of_star_mul_self ((star_star u).symm ▸ h_mul) |> unitary.star_mem)
 
+/-- In a star monoid, the product `a * b⁻¹` of units is unitary if `star a * a = star b * b`. -/
+lemma _root_.Units.mul_inv_mem_unitary (a b : Rˣ)
+    (hab : star a * a = star b * b) : (a * b⁻¹ : R) ∈ unitary R := by
+  apply IsUnit.mem_unitary_of_star_mul_self (by aesop)
```

Comment:
Is this true as an `Iff`? If so, does #24867 help?

Reply:
Yes, it can be upgraded to an `↔`, I will do it.

Reply:
Here's a nice proof:
```lean
lemma _root_.mul_inv_mem_unitary {G} [Group G] [StarMul G] (a b : G) :
    (a * b⁻¹) ∈ unitary G ↔ star a * a = star b * b := by
  rw [(Group.isUnit _).mem_unitary_iff_star_mul_self, star_mul, star_inv, mul_assoc,
    inv_mul_eq_iff_eq_mul, mul_one, ← mul_assoc, mul_inv_eq_iff_eq_mul]

theorem _root_.Units.unitary_eq : unitary Rˣ = (unitary R).comap (Units.coeHom R) := by
  ext
  simp [unitary.mem_iff, Units.ext_iff]

lemma _root_.Units.mul_inv_mem_unitary (a b : Rˣ) :
    (a * b⁻¹ : R) ∈ unitary R ↔ star a * a = star b * b := by
  rw [← _root_.mul_inv_mem_unitary, ← Units.val_mul, Units.unitary_eq]
  simp
```

Reply:
Though this makes me question whether the last lemma is needed at all.

Reply:
I'd prefer to keep it, as it's really the more common use case (i.e., most of the time, you're working with `unitary R` where `R` is a ring, so a monoid, not `unitary G` where `G` is a group).